can/interfaces/vector/canlib.py

```diff
@@ -93,6 +97,25 @@ def __init__(self, channel, can_filters=None, poll_interval=0.01,
         # Get channels masks
         self.channel_masks = {}
         self.index_to_channel = {}
+
+        if serialNumber is not None:
+            app_name = None
+            globalChannel = []
+            hwSpecificChannel = 0
+            hardwareConfigs = self._detect_available_configs()
```

Comment:
Perhaps it's best to create a new lower level method which yields `XLchannelConfig` that can be used by this code and `_detect_available_configs()`. That way you will have full access to all information.